build.gradle

```diff
@@ -28,12 +17,23 @@ dependencies {
     implementation gradleApi()
     implementation localGroovy()
 
-    compileOnly "com.buschmais.jqassistant.cli:jqassistant-commandline-neo4jv3:1.7.0"
+    compileOnly "com.buschmais.jqassistant.cli:jqassistant-commandline-neo4jv3:1.9.1"
 
     testImplementation gradleTestKit()
     testImplementation('org.junit.jupiter:junit-jupiter:5.5.2')
 }
 
+test {
+    useJUnitPlatform()
+    // add this plugin to maven local to allows using within the testProject
+    dependsOn("publishToMavenLocal")
```

Comment:
Hi f4lco, thanks for your feedback, I appreciate it. Unfortunately I'm currently a bit short of time, so that I'm not sure when I can rework the PR. Maybe you like to contribute, therefore I gave you access to the repo. Please feel free to change what every you think is appreciate. 

regards Alex

Reply:
Hi aews and  f4lco, thank you very much for your PR and comment. This plugin is my first one, I'm just a normal Gradle user. So if f4lco offers a review I appreciate that very much. Any comments (and PRs ;) ) are very welcome to make this plugin better. Indeed I'm currently a bit short of time, any help is very welcome.

Best regards,
Jens

Reply:
@aews thank you very much for giving me the liberty to do so. I appreciate that. Heads up: you seem to have a typo in your git email, it's pointing to "Alexander.Wessel@o-s,de". I fixed that for this PR.

@kontext-e sure I offer my help. For the scope of this PR I did the following:

- use `GradleRunner#withPluginClasspath`, which makes test setup easier: we can leave out the publication to mavenLocal, for example.
- I push for explicitly putting the "testProject" build file in the test resources, and to use a fresh temporary directory on each test run for reproducibility.
- Basing on ex "testProject", I created a new "example" project. Developers can try out new things in "example" without breaking tests. "example" also has a proper README. I really like the example rule @aews created. Using a composite build, Gradle reflects all changes to the plugin immediately in "example". 

I think that those changes are reasonable within the scope of this PR. Please feel free to review as time permits and to leave a thumbs up / down.

Reply:
@f4lco thanks for your help. Your changes looks good to me, so I m happy with it. Good job 👍